File: content/problems/0653-two-sum-iv-input-is-a-bst/solutions/dfs-complement-set/solution.cpp
class Solution {
    bool dfs(TreeNode* node,int target,unordered_set<int>& seen){
        if(!node)return false;
        if(seen.count(target-node->val))return true;
        seen.insert(node->val);
        return dfs(node->left,target,seen)||dfs(node->right,target,seen);
    }
    public:bool findTarget(TreeNode* root,int k){
        unordered_set<int> seen;
        return dfs(root,k,seen);
    }
};

File: content/problems/0653-two-sum-iv-input-is-a-bst/solutions/dfs-complement-set/Solution.kt
class Solution {
    fun findTarget(root:TreeNode?,k:Int):Boolean{
        val seen=hashSetOf<Int>();
        fun dfs(node:TreeNode?):Boolean{
            if(node==null)return false;
            if(k-node.`val` in seen)return true;
            seen.add(node.`val`);
            return dfs(node.left)||dfs(node.right)
        };
        return dfs(root)
    }
}
